### PR TITLE
[Unsigned]  Upgrade react script to v5

### PR DESCRIPTION
**Pull request by vipineth**
[x] Webpack 5 no longer includes node polyfills. Several dev dependencies needed to be added.
[x] There is no way to configure webpack in CRA without ejecting or using react-app-rewired.
[x] We are using `react-app-rewired`
[x] Change require to not use `default`

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1202793736284910

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -0,0 +1 @@
+GENERATE_SOURCEMAP=false
```
[the remark this answers is not in the document]
Comment (divhead) [reply .env:1]: Solved.
This does not seem to affect debugging code in dev mode

References:
https://github.com/facebook/create-react-app/pull/11752
https://github.com/facebook/create-react-app/discussions/11767

**Review by divhead — DISMISSED** [review]

File: .env (hunk 1 above)
[the remark this answers is not in the document]
Comment (vipineth) [reply .env:1]: Thanks for updating!

File: config-overrides.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
+const webpack = require("webpack");
+const ModuleScopePlugin = require("react-dev-utils/ModuleScopePlugin");
+
+module.exports = function override(config) {
+  config.resolve.plugins = config.resolve.plugins.filter((plugin) => !(plugin instanceof ModuleScopePlugin));
+  // https://github.com/lingui/js-lingui/issues/1195
+  config.module.rules[0].oneOf.splice(2, 0, {
```
Comment (divhead) [inline config-overrides.js:7]: Perhaps we should add a more detailed comment on what rule is changed here and why?

Or more explicitly, redefine this like:
config.module.rules[0] = {...} ?

Comment (vipineth) [reply config-overrides.js]: https://github.com/gmx-io/gmx-interface/pull/244/commits/33b99b68f2e270434cdbc3ffbe6d854487393ab5

Refactored and added comment explaining the loader we are adding to webpack ocnfig.

File: .env (hunk 1 above)
Comment (gdev8317) [inline .env:1]: will it have effect for local development?

Comment (vipineth) [reply .env:1]: No it only applies to production. You can check this link https://create-react-app.dev/docs/advanced-configuration/

**Pull request closed without merging**